runtime/recipe/view-connection.js

```diff
@@ -58,14 +60,29 @@ class ViewConnection {
   get recipe() { return this._recipe; }
   get name() { return this._name; } // Parameter name?
   get tags() { return this._tags; }
-  get type() { return this._type; }
+  get type() {
+    if (this._type)
+      return this._type;
+    return this._rawType;
+  }
+  get rawType() {
+    return this._rawType;
+  }
   get direction() { return this._direction; } // in/out
   get view() { return this._view; } // View?
   get particle() { return this._particle; } // never null
 
   set tags(tags) { this._tags = tags; }
-  set type(type) { this._type = type;}
-  set direction(direction) { this._direction = direction; }
+  set type(type) {
```

Comment:
`set rawType(...` -- I think we can be a bit more explicit about raw/resolved types.